source/UF/Size-TruncatedConnected.lagda

```diff
@@ -0,0 +1,417 @@
+
+\end{code}
+
+Local smallness is closed under equivalence, sigma types and truncation for each
+n : ℕ.
+
+\begin{code}
+
+local-smallness-is-closed-under-≃ : {X : 𝓤 ̇} {Y : 𝓦 ̇} {n : ℕ}
+                                  → X ≃ Y
+                                  → X is n locally-small
+                                  → Y is n locally-small
+local-smallness-is-closed-under-≃ {_} {_} {_} {_} {zero} e X-small =
+ smallness-closed-under-≃ X-small e
+local-smallness-is-closed-under-≃ {_} {_} {_} {_} {succ n} e X-loc-small y y' =
+ local-smallness-is-closed-under-≃ path-equiv (X-loc-small (⌜ e ⌝⁻¹ y) (⌜ e ⌝⁻¹ y'))
+ where
+  path-equiv : (⌜ e ⌝⁻¹ y ＝ ⌜ e ⌝⁻¹ y') ≃ (y ＝ y')
+  path-equiv = ≃-sym (ap ⌜ e ⌝⁻¹ , ap-is-equiv ⌜ e ⌝⁻¹ (⌜⌝⁻¹-is-equiv e))
+
+local-smallness-is-closed-under-Σ : {X : 𝓤 ̇} {Y : X → 𝓦 ̇} {n : ℕ}
+                                  → X is n locally-small
+                                  → ((x : X) → (Y x) is n locally-small)
+                                  → (Σ x ꞉ X , Y x) is n locally-small
+local-smallness-is-closed-under-Σ {_} {_} {_} {_} {zero} X-small Y-small =
+ Σ-is-small X-small Y-small
+local-smallness-is-closed-under-Σ {_} {_} {_} {Y} {succ n}
+ X-loc-small Y-loc-small (x , y) (x' , y') =
+ local-smallness-is-closed-under-≃ (≃-sym Σ-＝-≃)
+  (local-smallness-is-closed-under-Σ (X-loc-small x x')
+   (λ - → Y-loc-small x' (transport Y - y) y'))
+
+open general-truncations-exist te
+
+local-smallness-is-closed-under-truncation : {X : 𝓤 ̇} {n : ℕ₋₂}
+                                           → Univalence
+                                           → X is ι (n + 2) locally-small
+                                           → ∥ X ∥[ n ] is ι (n + 2) locally-small
+local-smallness-is-closed-under-truncation {_} {X} {−2} ua =
+ truncations-of-small-types-are-small
+local-smallness-is-closed-under-truncation {_} {X} {succ n} ua X-loc-small =
+ ∥∥ₙ-ind₂ (λ u v → (u ＝ v) is ι (n + 2) locally-small)
+          (λ u v → truncation-levels-are-upper-closed' ⋆
+                    (is-prop-implies-is-prop' (being-locally-small-is-prop ua)))
+          (λ x y → local-smallness-is-closed-under-≃
+                    (eliminated-trunc-identity-char (ua _))
+                   (local-smallness-is-closed-under-truncation ua (X-loc-small x y)))
+
+\end{code}
+
+Many of the results in [1] follow from a fact that some call the type theoretic
+axiom of replacement. It states that given a small type A, a locally small type
+X and a function f : A → X the image of f is small. In the paper 'the join
+construction' by Egbert Rijke, the axiom of replacement is shown to follow from
+the join construction. Currently, the join construction and the derivation of the
+axiom of replacement are not implemented in the TypeTopology library. We will
+state a more convenient but equivalent form of replacement, called
+Replacement': under the assumption that f : A → X is surjective (i.e. −1 connected)
```

Comment:
Currently it looks like what follows the `:` defines `Replacement'`. Please make clearer that what you're describing is why it's equivalent to replacement.